### PR TITLE
GCP compute instance public access 

### PR DESCRIPTION
**Pull request by umermehmood**

### CODE REVIEW

File: plugins/google/compute/instancePublicAccess.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        }
+                    }
+                });
+            });
+
+            if (myError[region] &&
+                zones[region] &&
+                (myError[region].join(',') === zones[region].join(','))) {
+                helpers.addResult(results, 3, 'Unable to query instances' , region);
+            } if (noInstances[region] &&
```
Comment (giorod3) [inline plugins/google/compute/instancePublicAccess.js:66]: is this supposed to be else if?

File: plugins/google/compute/instancePublicAccess.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                helpers.addResult(results, 0, 'No instances found in the region' , region);
+            } if (publicAccessInstances.length) {
+                var myInstanceStr = publicAccessInstances.join(", ");
+                helpers.addResult(results, 2,
+                    `Public access is enabled for following instances: ${myInstanceStr}`, region);
+            } if (privateInstances.length) {
+                var myInstanceStr = privateInstances.join(", ");
+                helpers.addResult(results, 0,
+                    `Public access is disabled for following instances: ${myInstanceStr}`, region);
+            }
```
Comment (giorod3) [inline plugins/google/compute/instancePublicAccess.js:78]: are these supposed to be else if's

Comment (umermehmood) [reply plugins/google/compute/instancePublicAccess.js:78]: Same reason goes here, multiple instances in one zone can have different settings. Let me know if there is anything I am missing.

File: plugins/google/compute/instancePublicAccess.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                if (!instances.data.length) {
+                    if (!noInstances[region]) {
+                        noInstances[region] = [];
+                    }
+                    noInstances[region].push(zone);
+                    return zcb();
+                }
+
+                instances.data.forEach(instance => {
+                    if (instance.networkInterfaces &&
```
Comment (AkhtarAmir) [inline plugins/google/compute/instancePublicAccess.js:48]: CIS says that instance names starting with 'gke-' should be skipped

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/google/compute/instancePublicAccess.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+var async   = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Instance Public Access Disabled',
+    category: 'Compute',
+    description: 'Ensures that compute instances are not configured to allow public access.',
+    more_info: 'Public IP address can cause security issues. To avoid the public access external ips should not be enabled.',
+    link: '',
```
Comment (AkhtarAmir) [inline plugins/google/compute/instancePublicAccess.js:9]: I removed this. Always use official google documentation

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**